packages/xod-func-tools/src/types.js

```diff
@@ -88,13 +88,22 @@ export const EnumType = R.curry(
   )
 );
 
+// For those who loves pain we have strict type checking mode that ignores
+// @@type hints and thus dead slow
+// :: Any -> Boolean
+const hasTypeHint = process.env.XOD_HM_DEF === 'strict' ? R.F : R.has('@@type');
+
 // Model :: String -> String -> String -> StrMap Type -> Type
 export const Model = R.curry(
   (packageName, docUrl, typeName, schema) => NullaryType(
     packageName,
     docUrl,
     typeName,
-    hasType($.RecordType(schema))
+    R.ifElse(
+      hasTypeHint,
```

Comment:
So, if we're in default(non-strict) mode, and a structure doesn't have a type hint, we just silently fall back to the slow check? Maybe, we should give some kind of warning in this case?